File: github-user-activity/Javascript/src/parser.ts
export const parseApiData = (event: any) => {
    const eventType = event["type"];
    switch (eventType) {
        case "PushEvent": {
            const repo = event["repo"]["name"];
            const ref = event["payload"]["ref"];
            let branch = "unknown branch";
            if (ref) {
                branch = ref.replace("refs/heads/", "");
            }
            return `Pushed commits to ${repo} (${branch})`;
        }
        case "WatchEvent": {
            const repo = event["repo"]["name"];
            return `Starred ${repo}`;
        }
        case "IssuesEvent": {
            const action = event["payload"]["action"];
            const repo = event["repo"]["name"];
            return `${action} an issue in ${repo}`;
        }
        case "PullRequestEvent": {
            const repo = event["repo"]["name"];
            const status = event["payload"]["action"];
            const sourceRepo = event["payload"]["pull_request"]["head"]["ref"];
            const targetRepo = event["payload"]["pull_request"]["base"]["ref"];
            if (status == "opened") {
                return `Created a Pull Requst from ${sourceRepo} to ${targetRepo} (${repo})`;
            }
            return `Pull Request from ${sourceRepo} to ${targetRepo} (${repo})`;
        }
        case "ForkEvent": {
            const repo = event["payload"]["forkee"];
            return `Forked ${repo}`;
        }
        case "CreateEvent": {
            const repo = event["repo"]["name"];
            const ref = event["payload"]["ref"];
            const refType = event["payload"]["ref_type"];
            if (refType == "branch") {
                return `Created ${refType} ${ref} in ${repo}`;
            }
            return `Created ${refType} ${repo}`;
        }
        case "DeleteEvent": {
            const refType = event["payload"]["ref_type"];
            const ref = event["payload"]["ref"];
            const repo = event["repo"]["name"];
            return `Deleted ${refType} ${ref} in ${repo}`;
        }
        default:
            return "Invalid Event type";
    }
};

File: github-user-activity/Python/github_activity/parser.py
def parseApiResponse(event):
        eventType = event['type']
        if eventType == 'PushEvent':
            repo = event['repo']['name']
            ref = event['payload']['ref']
            branch = ref.replace("refs/heads/","") if ref else "unknown branch"
            return f"Pushed commits to {repo} ({branch})"
        elif eventType == 'WatchEvent':
            repo = event['repo']['name']
            return f"Starred {repo}"
        elif eventType == 'IssuesEvent':
            action = event['payload']['action']
            repo = event['repo']['name']
            return f"{action} an issue in {repo}"
        elif eventType == 'PullRequestEvent':
            repo = event['repo']['name']
            status = event['payload']['action']
            sourceRepo = event['payload']['pull_request']['head']['ref']
            targetRepo = event['payload']['pull_request']['base']['ref']
            if status == 'opened':
                return f"Created a Pull Requst from {sourceRepo} to {targetRepo} ({repo})"
            return f"Pull Request from {sourceRepo} to {targetRepo} ({repo})"
        elif eventType == 'ForkEvent':
            repo = event['payload']['forkee']
            return f"Forked {repo}"
        elif eventType == 'CreateEvent':
            repo = event['repo']['name']
            ref = event['payload']['ref']
            refType = event['payload']['ref_type']
            if refType == 'branch':
                return f"Created {refType} {ref} in {repo}"
            else:
                return f"Created {refType} {repo}"
        elif eventType == 'DeleteEvent':
            refType = event['payload']['ref_type']
            ref = event['payload']['ref']
            repo = event['repo']['name']
            return f"Deleted {refType} {ref} in {repo}"
        else:
            print("Invalid Event Type")
